openfisca_core/entities.py

```diff
@@ -43,7 +43,14 @@ def get_variable(self, variable_name, check_existence = False):
         return self._tax_benefit_system.get_variable(variable_name, check_existence)
 
     def check_variable_defined_for_entity(self, variable_name):
-        variable_entity = self.get_variable(variable_name, check_existence = True).entity
+        try:
+            variable_entity = self.get_variable(variable_name, check_existence = True).entity
+        except AttributeError as e:
+            import sys
+            import traceback
+            traceback.print_tb(sys.exc_info()[2])
+            print(e)
+            raise ValueError("FIXME: tax_benefit_system is null")
```

Comment:
Update the error message by replacing "where the problem is" by "how you can solve the problem"?
You can find some information on the error message in the [contribution guide](https://github.com/openfisca/openfisca-core/blob/master/CONTRIBUTING.md#error-messages) of the repository.

Reply:
> If this error impacts the `/dependencies` endpoint and when we deliver it to the web API response, we have to keep in mind that the reader may not know precisely how openfisca is built (following the _principle of least astonishment_).

Reply:
The error says FIXME, because I wasn't sure if this is the right place to raise it.  I'm going to move it to the function above (where it originates) and push changes again. 

It shouldn't arise in the dependency API endpoint anymore, I had to set tax_benefit_system explicitly to mitigate it.

Please let me know if there's anything I've missed about the semantics of using the tax benefit system, as I hacked at it till it worked.